### ISSUE TITLE
Ubuntu 16.04 - key not found

### ISSUE THREAD

**Issue by colindaven** [ARTbio/GalaxyKickStart#237]
I can't get this key manually either.

Best, Colin

```
TASK [galaxyprojectdotorg.galaxy-os : Pin Galaxy's versions of the PPAs so they take priority over the defaults] ************************
skipping: [BioinformaticsWS]

TASK [galaxyprojectdotorg.galaxy-os : Add custom Galaxy PPA (used for nginx package)] ***************************************************
fatal: [BioinformaticsWS]: FAILED! => {"changed": false, "cmd": "apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 E9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED", "failed": true, "msg": "gpg: requesting key B9BDC5ED from hkp server keyserver.ubuntu.com\ngpg: no valid OpenPGP data found.\ngpg: Total number processed: 0\ngpg: keyserver communications error: keyserver helper general error\ngpg: keyserver communications error: unknown pubkey algorithm\ngpg: keyserver receive failed: unknown pubkey algorithm", "rc": 2, "stderr": "gpg: requesting key B9BDC5ED from hkp server keyserver.ubuntu.com\ngpg: no valid OpenPGP data found.\ngpg: Total number processed: 0\ngpg: keyserver communications error: keyserver helper general error\ngpg: keyserver communications error: unknown pubkey algorithm\ngpg: keyserver receive failed: unknown pubkey algorithm\n", "stderr_lines": ["gpg: requesting key B9BDC5ED from hkp server keyserver.ubuntu.com", "gpg: no valid OpenPGP data found.", "gpg: Total number processed: 0", "gpg: keyserver communications error: keyserver helper general error", "gpg: keyserver communications error: unknown pubkey algorithm", "gpg: keyserver receive failed: unknown pubkey algorithm"], "stdout": "Executing: /tmp/tmp.xPnKTkGbwA/gpg.1.sh --recv-keys\n--keyserver\nhkp://keyserver.ubuntu.com:80\nE9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED\ngpgkeys: key E9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED can't be retrieved\n", "stdout_lines": ["Executing: /tmp/tmp.xPnKTkGbwA/gpg.1.sh --recv-keys", "--keyserver", "hkp://keyserver.ubuntu.com:80", "E9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED", "gpgkeys: key E9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED can't be retrieved"]}
```

**Comment by mvdbeek** [comment]
Is that behind the proxy as well ? apt-key doesn't play nice in those circumstances.
Seems to work for me.
```
root@vm0089:~# apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 E9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED
Executing: gpg --ignore-time-conflict --no-options --no-default-keyring --homedir /tmp/tmp.OHWDn8Dfnt --no-auto-check-trustdb --trust-model always --keyring /etc/apt/trusted.gpg --primary-keyring /etc/apt/trusted.gpg --keyring /etc/apt/trusted.gpg.d/ansible-ansible.gpg --keyring /etc/apt/trusted.gpg.d/fkrull-deadsnakes.gpg --keyring /etc/apt/trusted.gpg.d/git-core-ppa.gpg --keyring /etc/apt/trusted.gpg.d/x2go-stable.gpg --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 E9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED
gpg: requesting key B9BDC5ED from hkp server keyserver.ubuntu.com
gpg: key B9BDC5ED: public key "Launchpad PPA for Marius van den Beek" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
```

**Comment by colindaven** [comment]
Bit of a nightmare but I found a fix here.


5
down vote
It seems that the proxy configuration is set up but somehow the server cannot be contacted...

I had a very similar issue which I resolved like this: The corporate proxy allows only port 80 and 443 for security reasons, so event when it is setup, since HKP protocol is using port 11371, it will not let you through.

So, short of SSH-ing out and getting the key from one of your servers outside, downloading it and installing it locally, you can specify the key server from on of the ones listed below and specify the port:

http://sks-keyservers.net/overview-of-pools.php

e.g.:

apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys XXXXX
Boom, Voila!

So this worked for me behind the proxy:

apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys E9708E5E0DEE60262A3FE2089EB12D87B9BDC5ED

**Comment by mvdbeek** [comment]
awesome, I had just ssh-ed out previously! We can add the p80 in the galaxy-os role.

**Comment by mvdbeek** [comment]
hmm, that doesn't work either for our proxy, but seems like a safe change.

**Comment by drosofff** [comment]
@colindaven with the last merged PR #236 I am not sure that you still need to add key manually. I you have some feedback to give I'll would appreciate !

**Comment by colindaven** [comment]
Pretty sure this worked without manual activity on my local PC. Thanks @drosofff
